pyomo/contrib/mindtpy/tests/test_mindtpy_solution_pool.py

```diff
@@ -0,0 +1,61 @@
+"""Tests for the MindtPy solver."""
+from pyomo.core.expr.calculus.diff_with_sympy import differentiate_available
+import pyomo.common.unittest as unittest
+from pyomo.contrib.mindtpy.tests.eight_process_problem import \
+    EightProcessFlowsheet
+from pyomo.contrib.mindtpy.tests.MINLP2_simple import SimpleMINLP as SimpleMINLP2
+from pyomo.contrib.mindtpy.tests.constraint_qualification_example import ConstraintQualificationExample
+from pyomo.environ import SolverFactory, value, maximize
+from pyomo.opt import TerminationCondition
+
+
+model_list = [EightProcessFlowsheet(convex=True),
+              ConstraintQualificationExample(),
+              SimpleMINLP2(),
+              ]
+
+required_solvers = ('ipopt', 'cplex_persistent', 'gurobi_persistent')
+if all(SolverFactory(s).available() for s in required_solvers):
+    subsolvers_available = True
+else:
+    subsolvers_available = False
```

Comment:
A test is feailing here with the following error:
``pyomo.common.errors.ApplicationError: No Python bindings available for <class 'pyomo.solvers.plugins.solvers.cplex_persistent.CPLEXPersistent'> solver plugin``
I would also suggest splitting the tests, sometimes CPLEX might be available but Gurobi not (or the opposite) so we want to run all the tests that we can.